### PR TITLE
readme rewrite.

### PR DESCRIPTION
**Pull request by tonyday567**
#491

I rewrote the readme, largely based on my own notes from recently learning about the project and introducing it for my libraries.

I have used travis-ci in the past, but couldn't find a modern haskell-ci travis usage example, so felt as if the detailed travis explanations could be dropped.

Happy to refine or go for a more surgical edit of the existing readme if any of this seems useful.

### CODE REVIEW

**Comment by tonyday567** [comment]
Also #522

**Comment by phadej** [comment]
Thanks. I skimmed through and it looks good, though there are some inaccuracies. I'll try to find some time to go through the new text (there is plenty).

File: README.md [hunk 1]
```diff
@@ -1,84 +1,66 @@
...
-    $ cat your-project.cabal
-    ...
-    Tested-With: GHC ==8.6.3 || ==8.4.4 || ==8.2.2
-    ...
-    ```
-    
-    Add as many or as few GHC versions to test as you want.
+``` shell
+haskell-ci travis '--config=haskell-ci.config' 'cabal.project' --output .travis.yml
+```
```
Comment (sjakobi) [inline README.md:46]: Is there a particular reason for specifying the default output paths via `--output`? I've never felt the need to use that option.

File: README.md [hunk 2]
```diff
@@ -1,84 +1,66 @@
...
-    using a `cabal.project` file, pass that file to haskell-ci instead:
     ```bash
-    $ haskell-ci cabal.project --output .travis.yml
+    $ git checkout master            # Check out `master`
+    $ git pull --ff-only             # Get the latest version of `master`
+    $ git checkout -b new_action     # Create a `new_action` branch
+    $ git add .github/workflows/haskell-ci.yml
+    $ git commit -m "New action"
+    $ git push -u origin new_action  # Push your branch upstream
     ```
```
Comment (sjakobi) [inline README.md:59]: This code block (and the next one) is malformatted:

image

Either indent the line above, so the code block is part of the numbered list, or un-indent the code block.

File: README.md [hunk 3]
```diff
@@ -1,84 +1,66 @@
...
-    $ git clone https://github.com/haskell-CI/haskell-ci.git
-    $ cd haskell-ci
-    $ cabal new-install haskell-ci:exe:haskell-ci
-    ```
+1. Install `haskell-ci`
 
-  or
+```bash
+cabal install haskell-ci
+```
```
Comment (sjakobi) [inline README.md:28]: Maybe say which GHC versions are supported for building `haskell-ci` or recommend a specific version here. Compare https://github.com/haskell-CI/haskell-ci/issues/557.

File: README.md [hunk 4]
```diff
@@ -1,84 +1,66 @@
...
- - GHC 7.8.1, GHC 7.8.2, GHC 7.8.3, GHC 7.8.4
- - GHC 7.10.1, GHC 7.10.2, GHC 7.10.3
- - GHC 8.0.1, GHC 8.0.2
- - GHC 8.2.1, GHC 8.2.2
- - GHC 8.4.1, GHC 8.4.2, GHC 8.4.3, GHC 8.4.4
- - GHC 8.6.1, GHC 8.6.2, GHC 8.6.3, GHC 8.6.4, GHC 8.6.5
- - GHC 8.8.1, GHC 8.8.2, GHC 8.8.3, GHC-8.8.4
- - GHC 8.10.1, GHC-8.10.2, GHC-8.10.3, GHC-8.10.4
- - GHC 9.0.1
- - GHC HEAD.
```
Comment (sjakobi) [inline README.md:21]: I think it would be useful to keep a list of the GHC versions that are supported.

**Review by sjakobi — COMMENTED** [review]
A few comments.

I think it would be good to make progress on this soon – I'm aware of several projects that have been trying to start using the GHA support in `haskell-ci` but struggled with the lack of documentation.

**Comment by phadej** [comment]
I trimmed README myself. I found this PR to invasive.

**Pull request closed without merging**